detox/src/devices/drivers/android/genycloud/helpers/GenyCloudInstanceLauncher.js

```diff
@@ -0,0 +1,36 @@
+const AndroidDeviceLauncher = require('../../AndroidDeviceLauncher');
+const GenyCloudInstanceHandle = require('../GenyCloudInstanceHandle');
+
+class GenyCloudInstanceLauncher extends AndroidDeviceLauncher {
+  constructor(instanceLifecycleService, deviceCleanupRegistry, eventEmitter) {
+    super(eventEmitter);
+    this._instanceLifecycleService = instanceLifecycleService;
+    this._deviceCleanupRegistry = deviceCleanupRegistry;
+  }
+
+  /**
+   * Note:
+   * In the context of Genymotion-cloud (as opposed to local emulators), the "launching"
+   * effectively only boils down to book-keeping associated with it, rather than the
+   * launch itself. That it because the actual launching (creation) of it is done in a synchronized,
+   * isolated means beforehand.
+   *
+   * @param instance {GenyInstance} The freshly allocated cloud-instance.
+   * @returns {Promise<void>}
+   */
+  async launch(instance) {
+    const instanceHandle = new GenyCloudInstanceHandle(instance);
+    await this._deviceCleanupRegistry.allocateDevice(instanceHandle);
+  }
+
+  async shutdown(instance) {
+    const instanceHandle = new GenyCloudInstanceHandle(instance);
```

Comment:
Generic question — aren't you trying here to make `detox test --cleanup` runs more favorable to Genycloud instead of planning a sprint where we move the device allocation/shutdown to global Jest setup and teardown?

In my opinion, `shutdown` should SHUT DOWN the device, and not do only book-keeping. Let's check if we are on the same page here with you, @d4vidi .

Reply:
Yes we will sort these things out sooner than later -- let's enqueue for the next Q! :)
In the meantime, the current code simply tries to comply to the currently expected behaviour, broken as it may be.